src/fluree/db/commit/storage.cljc

```diff
@@ -38,13 +38,29 @@
 (defn read-verified-commit
   [storage commit-address]
   (go-try
-    (when-let [commit-data (<? (storage/read-json storage commit-address))]
-      (log/trace "read-commit at:" commit-address "data:" commit-data)
-      (let [addr-key-path (if (contains? commit-data "credentialSubject")
-                            ["credentialSubject" "address"]
-                            ["address"])]
-        (-> commit-data
-            (assoc-in addr-key-path commit-address)
+    (when-let [commit-json (<? (storage/read-json storage commit-address))]
+      (log/trace "read-commit at:" commit-address "data:" commit-json)
+      (let [cred?         (contains? commit-json "credentialSubject")
+            subject-path  (if cred? ["credentialSubject"] [])
+            addr-path     (conj subject-path "address")
+            subject-json  (if cred? (get commit-json "credentialSubject") commit-json)
+            ;; Prefer commit id from address/hash; fallback to computing from JSON only if needed
+            existing-id   (get subject-json "id")
+            ;; Extract base32 hash-like token from the address if present
+            addr-str      (str commit-address)
+            hash-token    (last (clojure.string/split addr-str #"/"))
```

Comment:
There is a recently added [`get-hash` method on the `ContentArchive` protocol](https://github.com/fluree/db/blob/0ce80297e61f90cb59dffd4538b79e302fb48e60/src/fluree/db/storage.cljc#L130-L131)

Reply:
This was addressed in the upstream branch.